woocommerce/woocommerce-ios
# Create a Shipping Settings service for weight and dimension unit

Fixes #1507 

## Changes
- In Yosemite layer, created `ShippingSettingsService` protocol for getting dimension and weight units and updating when site ID changes
  - Implemented it with Core Data results controller in `StorageShippingSettingsService` + basic tests
- Added `ShippingSettingsService` to `ServiceLocator` + test case
- After syncing `SiteSetting`s, updated the site ID of `ShippingSettingsService`
- Shipping settings row on Add/Edit Product UI:
  - Displayed the weight info
  - Displayed the dimensions info

## Testing
- Launch the app
- Go to Products tab
- Tap on a Simple Product row where the weight and dimensions data are non-empty
- Tap "Edit" in the navigation bar --> should see the weight info and dimensions info
- Dismiss the Add/Edit Product screen
- Go back to Products tab
- Tap on a Simple Product row where the weight and dimensions data are empty
- Tap "Edit" in the navigation bar --> should just see grayed out "Shipping" on the shipping settings row
- Dismiss the Add/Edit Product screen

## Example screenshots

@kyleaparker we only display the dimensions info when the data for all 3 dimensions are non-empty right? So for example, if the product only has width info, we don't display the dimensions info

with weight and dimensions info | with just weight info | with just dimensions info | without weight and dimensions info
-- | -- | -- | --
![Simulator Screen Shot - iPhone 8 - 2019-11-27 at 13 15 39](https://user-images.githubusercontent.com/1945542/69693272-1eb09e00-1118-11ea-83da-a6185e118561.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-27 at 13 15 50](https://user-images.githubusercontent.com/1945542/69693275-1f493480-1118-11ea-8e2b-ef244f942867.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-27 at 13 16 03](https://user-images.githubusercontent.com/1945542/69693276-1f493480-1118-11ea-8dca-649fda438ac2.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-27 at 13 16 13](https://user-images.githubusercontent.com/1945542/69693279-1fe1cb00-1118-11ea-918a-a6d72c8c839e.png)

only 1 dimension available | only 2 dimensions available
-- | --
![Simulator Screen Shot - iPhone 8 - 2019-11-27 at 13 39 14](https://user-images.githubusercontent.com/1945542/69694594-abf5f180-111c-11ea-82c0-b148b381af44.png) | ![Simulator Screen Shot - iPhone 8 - 2019-11-27 at 13 46 05](https://user-images.githubusercontent.com/1945542/69694595-abf5f180-111c-11ea-8681-3f1e6b4afcc2.png)


Update release notes:
